packages/iobio-react-components/src/utils.ts

```diff
@@ -52,3 +62,68 @@ export const setElementStyles = (element: Element, styles: string) => {
 		element.shadowRoot.adoptedStyleSheets = [elementStyles];
 	}
 };
+
+/**
+ * Obtain BAM statistical data from Data Broker data events
+ * @param dataEvent { [BamKey]: number  }
+ */
+
+type StatisticsData = { [K in StatisticKey]: number };
+type PercentData = { [K in BamPercentKey]: number };
+
+type DataUpdate = StatisticsData & PercentData;
+
+export const getBamStatistics = (dataEvent: DataUpdate) => {
+	return [...percentKeys, ...statisticKeys].reduce(
+		(acc, val) => {
+			const value = dataEvent[val];
+			const stats: { [k: string]: number } = { ...acc, [val]: value };
+
+			if (percentKeys.some((percentKey) => percentKey === val)) {
+				const percentage = Number((value / dataEvent['total_reads']).toPrecision(4));
+				const key = `${val}_percentage`;
+				stats[key] = percentage;
+			}
+			return stats;
+		},
+		{} as { [k: string]: number },
+	);
+};
```

Comment:
would be good to explicitly name `(acc, val)` to what they are
for this looks like it's something like acc => stats and val => dataKey ?

Reply:
Yes explicit names are always preferred, will revise

Reply:
thanks! updated!